### ISSUE TITLE
Users other than Administrator Group cannot submit with form function

### ISSUE THREAD

**Issue by Toraji123** [pods-framework/pods#7096]
### Description

I am using [the form function](https://docs.pods.io/code/pods/form/) to allow users to create and update Custom Post Type data from the front end.

After upgrading Pods to 2.9.14, users other than the Administrator Group cannot submit with the form function.
To confirm, I rolled back to 2.9.13 and was able to submit it.

**The code where the error occurs:**
```
$my_pod = pods( 'my_custom_post_type' );
echo $my_pod->form();
```

**Error Message displayed after user submits form**
`Error: Pod not found`

![image](https://github.com/pods-framework/pods/assets/40485661/bd592ebb-ac60-41ce-8f24-d704d4114cf3)

This problem also occurs in 2.9.16.
My WordPress is 6.2.2. 

Another problem that occurs is that users other than the Administrator Group cannot view the Custom Post Type list. This also does not occur in 2.9.13.

2.9.13
![image](https://github.com/pods-framework/pods/assets/40485661/ce540cdc-499f-4c77-9ab0-6987fd45c397)

2.9.14 or later
![image](https://github.com/pods-framework/pods/assets/40485661/5b714d9a-eb9a-4267-a6c4-1a5e3af22771)

### Version

2.9.14

### Testing Instructions

1. Create a Custom Post Type.
2. Write the form function for the Custom Post Type in the theme.
```
$my_pod = pods( 'my_custom_post_type' );
echo $my_pod->form();
```

1. Log in as a user other than the Administrator Group (like Editor).
2. Submit the form.

**Comment by sc0ttkclark** [comment]
I can't reproduce this issue, we can reopen this if anyone can confirm this is happening on 2.9.19+

**Comment by Toraji123** [comment]
@sc0ttkclark 
Thank you for your investigation.
This problem seems to have been resolved in 2.9.19.
